docker-entrypoint.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+set -e
+/usr/local/bin/ptunnel-ng "$@" &
+wait
```

Comment:
I'm just wondering why you send ptunnel-ng to the background? Do you want bring it to the foreground later on? 
I'm not a docker pro so I might miss some information here.

Reply:
When killing a Docker container, Docker will send `SIGKILL` to PID 1 inside the container. `ptunnel-ng` doesn't seem to respond properly to this signal (possibly due to `setuid()`), so I included this wrapper script to launch it as a subprocess, allowing for graceful exit. 
Maybe not the cleanest way of doing things but it works.

Reply:
AFAIK, `SIGKILL` can't intercepted from userspace applications (at least not by a signal handler) and `setuid` should not interfere with signal handling.
I need to investigate what is going on here. Thanks for your answer.

Reply:
Small correction to the above: Hitting Ctrl+C will send `SIGINT`, not `SIGTERM`, to PID 1 in the attached container.

Here is a decent explanation of why this doesn't work: https://engineeringblog.yelp.com/2016/01/dumb-init-an-init-for-docker.html

Basically, unless signal handling is explicitly specified in the program, running a normal (non-init) process as PID 1 inside a container causes `SIGINT` to be deflected. This is by design of the Linux kernel.

I've modified slightly the Dockerfile to use a real init common in container environments called `tini` to handle this signalling issue.

A good explanation from the `tini` authors about why to use an init binary: https://github.com/krallin/tini/issues/8#issuecomment-146135930

Reply:
It worked like a charm. Thanks for your contribution!